src/chapters/angular-lsn3/exercises.rst

```diff
@@ -11,31 +12,250 @@ The starter code for the exercises is in the same
+      <img #rocketImage src="assets/images/LaunchCode_rocketline_white.png" height = "75" width = "75" [style.left]="0" [style.bottom]="0"/>
+
+2. While you're here, add the click handler to the *Right* button:
+
+.. sourcecode:: html+ng2
+      
+      <button (click)="moveRocket(rocketImage, 'right')">Right</button>
+
+3. Now in ``app.component.ts`` we can write the ``handleRightClick()``:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   moveRocket(rocketImage, direction) {
+    if (direction === 'right') {
+      let movement = parseInt(rocketImage.style.left) + 10 + 'px';
+      rocketImage.style.left = movement;
+      this.width = this.width + 10000;
+    } 
+   }
+
+4. Add conditional logic to this ``moveRocket()`` method to account for the other movement directions, modifying the
+movement formula as needed. Be sure to also update the height or width property
+where appropriate.
+
+Update the Control Button Click Handlers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Along those same lines, we'll want to modify a couple of our control 
+button handlers to update ``rocketImage``'s position when the status
+changes. Pass in ``rocketImage`` to your *Land* and *Abort Mission* 
+handlers and add the following:
+
+.. sourcecode:: TypeScript
+
+   rocketImage.style.bottom = '0px';
+
+New Requirements
 ----------------
+
+1. Right now, a user can move the rocket before it has officially taken off or abort the 
+mission while the rocket is still on the ground. This doesn't make much sense. With attribute
+directives, we can dynamically set those buttons to only be enabled in some states.
+
+Let's add a check for the take off status of the shuttle.
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+    takeOffEnabled: true,
+    
+
+2. When the app is first loaded, we want the user to be able to click the *Take Off*
+button, but not the *Land* or *Abort Mission* button. We'll add some 
+``[disabled]`` attribute directives on the control buttons to reflect these values.
+
+Update the control buttons:
+
+.. sourcecode:: html+ng2
+   :linenos:
+      
+   <div class="container-control-buttons">
+      <button (click)="handleTakeOff()" [disabled]="!takeOffEnabled">Take Off</button>
+      <button (click)="handleLand(rocketImage)" [disabled]="takeOffEnabled">Land</button>
+      <button (click)="handleMissionAbort(rocketImage)" [disabled]="takeOffEnabled">Abort Mission</button>
+   </div>
+
+Now, based on the boolean ``takeOffEnabled``, only the *Take Off* control button 
+is enabled when the rocket is on the ground.
+
+Update the control button click handlers to toggle the enabled/disabled status 
+of the controls using this value.
+
+3. For another improvement, we shouldn't be able to move the rocket if it hasn't taken off. To toggle the status of
+the direction buttons, we could add more boolean checks to our component. However, we know we only
+want these buttons to be accessible when the *Take Off* button is not. We can therefore take advantage
```

Comment:
Rephrasing suggestion: "We can therefore re-use ``takeOffEnabled`` to determine if the user can click the direction buttons."